lib/sapic/src/Sapic.hs

```diff
@@ -111,14 +111,13 @@ translate th = case theoryProcesses th of
                                                                  -- else
                                                                  --   restrs
                                                                  --    @ (if op.progress then [progress_init_lemma] else [])
-                        $ [BT.baseRestr anP needsAssImmediate (containChannelIn pRules) True] ++
```

Comment:
Hi Kevin, what's the reason to give the annotated rules as the parameter to `restrictions` and not the outcome `containChannelIn protoRules`?  I can see that `BT.baseRestr` needs to know if a ChannelIn appears and that this is somehow specific to the base translation. But in that case, `containChannelIn` should be in `BaseTranslation`. Even better would be to make this check in `baseRestr` on the annotated process. It would preserve the existing structure and put the implementation-specific check (containChannelIn, but on annotated processes) in the appropriate module.